src/@ionic-native/plugins/intel-security/index.ts

```diff
@@ -0,0 +1,248 @@
+import { Plugin } from '@ionic-native/core';
+import { Injectable } from '@angular/core';
+
+declare var window: any;
+declare var intel: any;
+
+/**
+ * @name Intel Security
+ * @description
+ * The App Security API enables the use of security properties and capabilities on the platform, using a new set of API defined for application developers. You are not required to be a security expert to make good use of the API. Key elements, such as encryption of data and establishments of capabilities, is abstracted and done by the API implementation, for you.
+ *
+ * For example:
+ * - Use the API to store (E.g. cache) data locally, using the device non-volatile storage. Data protection/encryption will be done for you by the API implementation
+ * - Establish a connection with remote server (E.g. XHR) using a protected channel. SSL/TLS establishment and usage will be done for you by the API implementation
+ *
+ * For more information please visit the [API documentation](https://software.intel.com/en-us/app-security-api/api).
+ *
+ * @usage
+ * ```
+ * import { IntelSecurity } from '@ionic-native/intel-security';
+ * ...
+ * constructor(private intelSecurity: IntelSecurity) { }
+ * ...
+ *
+ * let storageID = 'id';
+ *
+ * this.intelSecurity.data.createFromData('Sample Data')
+ *   .then((instanceID: any) => this.IntelSecurity.storage.write(id, instanceID))
+ *   .catch((error: any) => console.log(error));
+ *
+ * this.intelSecurity.storage.read(storageID)
+ *   .then(this.intelSecurity.data.getData)
+ *   .then((data: string) => console.log(data)) // Resolves to 'Sample Data'
+ *   .catch((error: any) => console.log(error));
+ *
+ * this.intelSecurity.storage.delete(storageID)
+ *   .then(() => console.log('Deleted Successfully'))
+ *   .catch((error: any) => console.log(error));
+ *
+ * ```
+ */
+@Plugin({
+  pluginName: 'IntelSecurity',
+  plugin: 'com-intel-security-cordova-plugin',
+  pluginRef: 'intel.security',
+  repo: 'https://github.com/AppSecurityApi/com-intel-security-cordova-plugin'
+})
+@Injectable()
+export class IntelSecurity {
+
+  public storage: IntelSecurityStorage;
+  public data: IntelSecurityData;
+
+  constructor() {
+    this.storage = new IntelSecurityStorage();
+    this.data = new IntelSecurityData();
+
+    try {
+      if (!window.intel)
+        throw true;
+    } catch (e) {
+      console.warn('Native: Intel App Security API is not installed or you are running on a browser.');
+    }
+  }
+}
+
+/**
+ * @hidden
+ */
+export class IntelSecurityData {
```

Comment:
You can add another `@Plugin()` tag here with a different `pluginRef` and no `repo` property. You can also make it injectable if you think that's a better way to do it.

If you make it Injectable. Then the class above should be empty and only used for documentation purposes. 

---

Apply this fix to the class below as well.